lib/components/user/places/place-editor.js

```diff
@@ -0,0 +1,124 @@
+const customPlaces = clone(PLACE_TYPES)
+delete customPlaces.home
+delete customPlaces.work
+
+/**
+ * Contains the fields for editing a favorite place.
+ * This component uses Formik props that are passed
+ * within the Formik context set up by FavoritePlaceScreen.
+ */
+class PlaceEditor extends Component {
+  _handleLocationChange = ({ location }) => {
+    const { setValues, values } = this.props
+    const { lat, lon, name } = location
+    setValues({
+      ...values,
+      address: name,
+      lat,
+      lon
+    })
+  }
+
+  render () {
+    const { errors, handleBlur, handleChange, values: place } = this.props
+    const isFixed = isHomeOrWork(place)
+    const errorStates = getErrorStates(this.props)
+
+    return (
+      <div style={{marginTop: '40px'}}>
+        {!isFixed && (
+        <>
+          <FormGroup validationState={errorStates.name} >
+            {/* onBlur, onChange, and value are passed automatically. */}
+            <Field as={FormControl}
+              aria-label={MESSAGES.SET_PLACE_NAME}
+              name='name'
+              placeholder={MESSAGES.SET_PLACE_NAME}
+            />
+            <FormControl.Feedback />
+            {errors.name && <HelpBlock>{errors.name}</HelpBlock>}
+          </FormGroup>
+
+          <FormGroup>
+            <ToggleButtonGroup
+              name='type'
+              type='radio'
+              defaultValue={place.type}
+            >
+              {Object.values(customPlaces).map(({ icon, text, type }) => (
+                <ToggleButton
+                  key={type}
+                  // onBlur and onChange have to be set on individual controls instead of the control group
+                  // in order for Formik to correctly process the changes.
+                  onBlur={handleBlur}
+                  onChange={handleChange}
+                  title={text}
+                  value={type}
+                >
+                  <LargeIcon name={icon} />
+                </ToggleButton>
+              ))}
+            </ToggleButtonGroup>
+          </FormGroup>
+        </>
+        )}
+
+        <div style={{ display: 'flex' }}>
+          {/* For fixed places, just show the icon for place type instead of all inputs and selectors */}
+          {isFixed && <FixedPlaceIcon name={place.icon} />}
+
+          <FormGroup style={{flexGrow: 1}} validationState={errorStates.address}>
+            <PlaceLocationField // TODO: mobile version.
+              className='form-control'
+              inputPlaceholder={isFixed ? `Search for ${place.name} location` : 'Search for location'}
+              location={makeLocationFieldLocation(place)}
+              locationType='to'
+              onLocationSelected={this._handleLocationChange}
+              showClearButton={false}
+            />
+            <FormControl.Feedback />
+            {errors.address && <HelpBlock>{errors.address}</HelpBlock>}
+
+            {/* TODO: Implement setting location using map. */}
```

Comment:
I'm not a fan of leaving TODO comments in code because there is no guarantee these ideas are ever implemented. If it's really important we can make an issue, or perhaps better yet if it truly is needed we will just remember about it later without having created an issue or writing a code comment.

Reply:
Removed in https://github.com/opentripplanner/otp-react-redux/pull/280/commits/53797da17e0d91c1c07659bbd169f4e14ec8683f.